File: src/test/java/com/devon/dapuzzo/EmailRequestTest.java
package com.devon.dapuzzo;

import org.junit.Test;

import static org.junit.Assert.*;

/**
 * Created by devondapuzzo on 9/7/17.
 */
public class EmailRequestTest {

    @Test
    public void shouldVerifyValidEmailAddress() throws Exception {
        EmailRequest emailRequestGood = new EmailRequest("joe", "joe@gmail.com", "Hey bro");
        assertTrue(emailRequestGood.isValid());

        EmailRequest emailRequestBad = new EmailRequest();
        assertFalse(emailRequestBad.isValid());

        emailRequestBad = new EmailRequest("joe", "joe@gmailcom", "Hey bro");
        assertFalse(emailRequestBad.isValid());
    }
}

File: src/test/java/com/devon/dapuzzo/email/EmailRequestTest.kt
package com.devon.dapuzzo.email

import com.devon.dapuzzo.util.random.randomEmailRequest
import com.devon.dapuzzo.util.random.randomString
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test


/**
 * Created by devondapuzzo on 9/7/17.
 */
class EmailRequestTest {

    @Test
    fun `should verify valid email address`() {
        val subject = randomEmailRequest()
        assertThat(subject.isValid).isTrue()
    }

    @Test
    fun `should reject malformed email address`() {
        val subject = EmailRequest(randomString(), randomString(), randomString())
        assertThat(subject.isValid).isFalse()
    }
}